v1.0.0.yaml

```
@@ -2952,203 +2823,913 @@ paths:
 
-                    Note: When passing a URL in the file parameter, please ensure that our servers can access the URL. In case ImageKit is unable to download the file from the specified URL, a 400 error response is returned. In addition to this, the file download request is aborted if response headers are not received in 8 seconds. This will also result in a `400` error.
-                  example: https://www.example.com/rest-of-the-image-path.jpg.
-                fileName:
-                  type: string
-                  description: |
-                    The name with which the file has to be uploaded.
 
-                    The file name can contain:
+            **Note**: Sending a value that has been used in the past will result in a validation error. Even if your previous request resulted in an error, you should always send a new value for this field.
+        useUniqueFileName:
+          type: string
+          default: true
+          description: >
+            Whether to use a unique filename for this file or not.
+            
+            
+            If `true`, ImageKit.io will add a unique suffix to the filename parameter to get a unique filename.
+            
+            
+            If `false`, then the image is uploaded with the provided filename parameter, and any existing file with the same name is replaced.
+          enum:
+            - true
+            - false
+        tags:
+          type: string
+          description: >
+            Set the tags while uploading the file.
+            
+            
+            Comma-separated value of tags in the format `tag1,tag2,tag3`.
+            The maximum length of all characters should not exceed 500. `%` is not allowed.
+            
+            
+            If this field is not specified and the file is overwritten then the tags will be removed.
+          example: "t-shirt,round-neck,men"
+        folder:
+          type: string
+          description: >
+            The folder path in which the image has to be uploaded.
+            If the folder(s) didn't exist before, a new folder(s) is created.
+            
+            
+            The folder name can contain:
+            
+              - Alphanumeric Characters: `a-z` , `A-Z` , `0-9`
+              - Special Characters: `/` , `_` , `-`
+            
+            Using multiple `/` creates a nested folder.
+          default: "/"
+        isPrivateFile:
+          type: string
+          default: false
+          description: >
+            Whether to mark the file as private or not.
+            
+            
+            If `true`, the file is marked as private and is accessible only using named transformation or signed URL.
+          enum:
+            - true
+            - false
```

The enum & default values are of the type `boolean` while the parameter is of the type `string`. This might show up as errors for users in some tools.

```suggestion
        isPrivateFile:
          type: string
          default: "false"
          description: >
            Whether to mark the file as private or not.
            
            
            If `true`, the file is marked as private and is accessible only using named transformation or signed URL.
          enum:
            - "true"
            - "false"
```

Other boolean-string parameters have the same issue.